maven-core/src/main/java/org/apache/maven/execution/ProfileActivationRequest.java

```diff
@@ -0,0 +1,180 @@
+package org.apache.maven.execution;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * Request from the user to activate or de-activate certain profiles and optionally fail the build if those profiles
+ * do not exist.
+ */
+public class ProfileActivationRequest
```

Comment:
I would pick a different classname. In general if there is a `request` I would expect a `result` too.

Reply:
Fair point. I guess `ProfileActivation` would make more sense either way. I'll take a look at this tomorrow.

Reply:
Also keep the modules in mind, they'll need a similar solution

Reply:
Yup, currently the respective class for project selection is called `ProjectActivation`, so I think `ProfileActivation` is consistent.

@mthmulders and I have started working on the 'optional' project activation issue, by the way (https://issues.apache.org/jira/browse/MNG-6511).